src/tools/remote/src/main/java/com/google/devtools/build/remote/worker/ExecutionServer.java

```diff
@@ -151,11 +151,7 @@ public void waitExecution(WaitExecutionRequest wr, StreamObserver<Operation> res
       return;
     }
     ((ServerCallStreamObserver<Operation>) responseObserver)
-        .setOnCancelHandler(
-            () -> {
-              future.cancel(true);
```

Comment:
Why is this ok to remove? (I'm not that familiar with workers).

Reply:
I added this in https://github.com/bazelbuild/bazel/commit/ba8ccaf48a911f7bc9f2c18ab8f2fdb71d5c64ac. The call to `setOnCancelHandler` is necessary to deflake the test, but calling `future.cancel(true)` has the unwanted effect that a single failing action executed by the worker stops all other concurrently running actions, which then causes the consuming Bazel instance to fail with a connection error rather than report the failing actions as such.